File: app/src/main/java/com/shmedo/mcloudapp/deviceconfig/adapter/ProductAdapter.java
package com.shmedo.mcloudapp.deviceconfig.adapter;

import android.graphics.Typeface;
import android.widget.TextView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.viewholder.BaseViewHolder;
import com.shmedo.mcloudapp.R;
import com.shmedo.mcloudapp.deviceconfig.model.ProductInfo;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

/**
 * 创建者:   gonghe <br/>
 * 创建时间:  2020/8/18 <br/>
 * 描述：    产品适配器
 */
public class ProductAdapter extends BaseQuickAdapter<ProductInfo, BaseViewHolder> {

    public ProductAdapter(@Nullable List<ProductInfo> data) {
        super(R.layout.item_product, data);
    }

    @Override
    protected void convert(@NotNull BaseViewHolder holder, ProductInfo productInfo) {
        holder.setText(R.id.tv_name, productInfo.getProductName());
        if (productInfo.getProductName().equals("全部")) {
            holder.setGone(R.id.tv_num, true);
        } else {
            holder.setGone(R.id.tv_num, true);
//            holder.setText(R.id.tv_num, "(" + productInfo.getDeviceNum() + ")");
        }
        TextView tvName = (TextView) holder.getView(R.id.tv_name);
        TextView tvNum = (TextView) holder.getView(R.id.tv_num);
        if (productInfo.isChecked()) {
            holder.setBackgroundResource(R.id.ll_item, R.drawable.bg_device_type_checked);
            holder.setTextColorRes(R.id.tv_name, R.color.blue_52B4F8);
            tvName.setTextSize(17);
            tvNum.setTextSize(17);
            tvName.setTypeface(tvName.getTypeface(), Typeface.BOLD);
        } else {
            holder.setBackgroundResource(R.id.ll_item, 0);
            holder.setTextColorRes(R.id.tv_name, R.color.text_color_666666);
            tvName.setTextSize(15);
            tvNum.setTextSize(15);
            tvName.setTypeface(tvName.getTypeface(), Typeface.NORMAL);
        }
    }
}

File: app/src/main/java/com/shmedo/mcloudapp/common/adapter/ProductAdapter.kt
package com.shmedo.mcloudapp.common.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.viewholder.BaseDataBindingHolder
import com.shmedo.lib.core.base.model.ProductInfo
import com.shmedo.mcloudapp.R
import com.shmedo.mcloudapp.databinding.ItemProductBinding

/**
 * 创建者：gonghe
 *
 * 创建时间：2023/8/31
 *
 * 描述： TODO
 *
 *
 */
class ProductAdapter : BaseQuickAdapter<ProductInfo, BaseDataBindingHolder<ItemProductBinding>>(
    R.layout.item_product
) {
    override fun convert(holder: BaseDataBindingHolder<ItemProductBinding>, item: ProductInfo) {
        // 获取 Binding
        val binding = holder.dataBinding
        if (binding != null) {
            onBindItem(binding, item)
            binding.executePendingBindings()
        }
    }

    private fun onBindItem(
        binding: ItemProductBinding,
        item: ProductInfo
    ) {
        // 设置数据
        binding.m = item
        if (item.isChecked)
            binding.tvName.setTextAppearance(R.style.Product_Tag_Checked_TitleStyle)
        else
            binding.tvName.setTextAppearance(R.style.Product_Tag_UnChecked_TitleStyle)
    }
}